tdms/src/iterator.cpp

```diff
@@ -2741,60 +2702,60 @@ void mexFunction(int nlhs, mxArray *plhs[], int nrhs, const mxArray *prhs[])
   t0 = (double)time(NULL);
   //    fprintf(stderr,"start_tind: %d\n",start_tind);
   //  fprintf(stdout,"dz: %e, c: %e, dz/c: %e\n",dz,light_v,dz/light_v);
-  //Begin of main iteration loop 
-  if(TIME_MAIN_LOOP)
-    time_ml_0 = omp_get_wtime();
+  //Begin of main iteration loop
+  auto main_loop_timer = Timer();
+
+  if(TIME_MAIN_LOOP){
+    main_loop_timer.start();
+  }
+
   for(tind = start_tind; tind < *Nt; tind++){
     //fprintf(stderr,"Pos 00:\n");
     time_E = ( (double)(tind + 1) )*dt[0];
     time_H = time_E - *dt/2.;
     //Extract phasors
-    time_0=omp_get_wtime();
+    auto timer = Timer();
     if((dft_counter == Nsteps)&&(runmode==rm_complete)&&(sourcemode==sm_steadystate) && exphasorsvolume){//runmode=complete,sourcemode=steadystate
       dft_counter = 0;
-      double tol = checkPhasorConvergence(ExR,ExI,EyR,EyI,EzR,EzI,
-					  ExR2,ExI2,EyR2,EyI2,EzR2,EzI2,
-					  Exy,Exz,Eyx,Eyz,Ezx,Ezy,
-					  pind_il,pind_iu,
-					  pind_jl,pind_ju,
-					  pind_kl,pind_ku,
-					  tind,*omega_an, *dt, *Nt);
-      
-      
+      double tol = checkPhasorConvergence(E, E_copy, E_s,
+                                          pind_il, pind_iu,
+                                          pind_jl, pind_ju,
+                                          pind_kl, pind_ku,
+                                          tind, *omega_an, *dt, *Nt);
+
+
       //      mexPrintf("tol: %.5e \n",tol);
       fprintf(stderr,"tol: %.5e \n",tol);
-      
+
       if( tol<TOL )
 	break;//required accuracy obtained
-      
-      copyPhasors(mxGetPr( (mxArray *)plhs[0]),mxGetPi( (mxArray *)plhs[0]),mxGetPr( (mxArray *)plhs[1]),mxGetPi( (mxArray *)plhs[1]),mxGetPr( (mxArray *)plhs[2]),mxGetPi( (mxArray *)plhs[2]),
-		  mxGetPr( (mxArray *)dummy_array[0]),mxGetPi( (mxArray *)dummy_array[0]),mxGetPr( (mxArray *)dummy_array[1]),mxGetPi( (mxArray *)dummy_array[1]),
-		  mxGetPr( (mxArray *)dummy_array[2]),mxGetPi( (mxArray *)dummy_array[2]),  (int)mxGetNumberOfElements( (mxArray *)plhs[0]));
-      
+
+      copyPhasors(E, E_copy, (int)mxGetNumberOfElements( (mxArray *)plhs[0]));

```

Comment:
not entirely sure of this change – _possibly_ copies in a different order but someone with more experience with pointers should check this!